app/javascript/components/modals/ExerciseCompletedModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { Modal } from './Modal'
+import { GraphicalIcon } from '../common'
+import { ConceptProgression } from './exercise-completed-modal/ConceptProgression'
+import { UnlockedExercise } from './exercise-completed-modal/UnlockedExercise'
+import { UnlockedConcept } from './exercise-completed-modal/UnlockedConcept'
+import { ExerciseCompletion } from './CompleteExerciseModal'
+import pluralize from 'pluralize'
+
+export const ExerciseCompletedModal = ({
+  open,
+  completion,
+  ...props
+}: {
+  open: boolean
+  completion: ExerciseCompletion
+}): JSX.Element => {
+  const {
+    exercise,
+    conceptProgressions,
+    unlockedExercises,
+    unlockedConcepts,
+  } = completion
+
+  return (
+    <Modal
+      open={open}
+      className="m-completed-exercise"
+      onClose={() => {}}
+      {...props}
+    >
+      <header>
+        <GraphicalIcon icon={exercise.iconName} className="c-exercise-icon" />
+        <div className="exercise-title">
+          You&apos;ve completed
+          <br />
+          {exercise.title}!
+        </div>
+      </header>
+      <div className="progressed-concepts">
+        {conceptProgressions.map((progression) => (
+          <ConceptProgression key={progression.name} {...progression} />
+        ))}
+      </div>
+      <div className="unlocks">
```

Comment:
This whole `.unlocks` div shouldn't show if there are no unlocked exercises and no unlocked concepts.